app/Svg/Parser.hs

```diff
@@ -71,10 +71,19 @@ performParse :: T.Text -- ^ The title of the graph.
              -> String -- ^ The filename of the file that will be parsed.
              -> SqlPersistM ()
 performParse graphName inputFilename = do
+    deleteExistingGraph graphName
     liftIO . print $ "Parsing graph " ++ T.unpack graphName ++ " from file " ++ inputFilename
     graphFile <- liftIO $ T.readFile (graphPath ++ inputFilename)
     performParseFromMemory graphName graphFile False
 
+-- | Check if a graph exists and delete it if it does
+deleteExistingGraph :: T.Text -> SqlPersistM ()
+deleteExistingGraph graphName = do
+  maybeResponse <- liftIO $ getGraph graphName
+  case maybeResponse of
+    Just _ -> deleteGraph graphName
```

Comment:
After doing what I've said above, you don't need to pass in the `graphName` here; instead, you should be able to pass in the graph Id itself, so that the type of `deleteGraph` becomes something like `Key Graph -> SqlPersistM ()`.